packages/utils/src/bulk-update-patients.ts

```diff
@@ -0,0 +1,192 @@
+const sqlDBCreds = getEnvVarOrFail("DB_CREDS"); // Must use the read replica
+const cxId = getEnvVarOrFail("CX_ID");
+const CHUNK_DELAY_MAX_MS = dayjs.duration({ minutes: 1 }).asMilliseconds();
+const PATIENT_CHUNK_SIZE = 5;
+
+const patientCreatedDate = ""; // YYYY-MM-DD format
+
+const metriportAPI = new MetriportMedicalApi(apiKey, {
+  baseAddress: apiUrl,
+});
+
+const cqPatientDataResultSchema = z.array(
+  z.object({
+    id: z.string(),
+    cx_id: z.string(),
+    data: z.object({
+      links: z.array(z.object({ oid: z.string(), url: z.string() })),
+    }),
+    created_at: z.date(),
+  })
+);
+type CqPatientDataResult = z.infer<typeof cqPatientDataResultSchema>;
+
+async function main() {
+  const dbCreds = JSON.parse(sqlDBCreds);
+  if (!patientCreatedDate) {
+    console.log("Please provide a date string in the format YYYY-MM-DD");
+    return;
+  }
+  const targetDate = new Date(patientCreatedDate);
+
+  const sequelize = new Sequelize(dbCreds.dbname, dbCreds.username, dbCreds.password, {
+    host: dbCreds.host,
+    port: dbCreds.port,
+    dialect: dbCreds.engine,
+  });
+
+  try {
+    const facilities = await metriportAPI.listFacilities();
+    await displayInitialWarningAndConfirmation(facilities.length);
+    const patientCQLinks = await getPatientCqLinks(sequelize, cxId);
+
+    let totalFacilities = 0;
+    let totalPatients = 0;
+    let patientsNoLinks = 0;
+
+    for (let i = 0; i < facilities.length; i++) {
+      const facility = facilities[i];
+      totalFacilities++;
+      const patientsList = await metriportAPI.listPatients(facility.id);
+
+      await displayWarningAndConfirmation(facilities.length);
+      const patientsCreatedAfter = patientsList.filter(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        patient => new Date(patient.dateCreated!) > targetDate
+      );
+
+      totalPatients += patientsCreatedAfter.length;
+
+      const patientsWithNoLinks = [];
+
+      for (const patient of patientsCreatedAfter) {
+        const hasPatientLinks = patientCQLinks.filter(link => link.id === patient.id);
+        if (hasPatientLinks.length > 0) {
+          continue;
+        }
+
+        patientsWithNoLinks.push(patient);
+        patientsNoLinks++;
+      }
+
+      const patientChunks = chunk(patientsWithNoLinks, PATIENT_CHUNK_SIZE);
+      console.log(`Facility ${facility.id} has ${patientsWithNoLinks.length} patients`);
+
+      for (const [j, patients] of patientChunks.entries()) {
+        console.log(`Chunk ${j + 1} of ${patientChunks.length}`);
+        console.log(`# of patients ${patients.length}`);
+
+        for (const patient of patients) {
+          console.log(`Updating patient ${patient.id}`);
+          const addressObject = Array.isArray(patient.address)
+            ? patient.address[0]
+            : patient.address;
+          const address = addressObject as Address;
+          await updatePatient(patient, address, facility.id);
```

Comment:
@thomasyopes is adding multiple addresses, we should take that into account.

Also, this would remove any address that we augmented from links.

We should probably write an internal endpoint to which we could send a list of patient IDs and it would trigger a PD instead of updating the patient using the SDK.